internal/controllers/cluster/cluster_controller.go

```diff
@@ -314,28 +347,18 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Clu
 	}
 
 	if cluster.Spec.ControlPlaneRef != nil {
-		obj, err := external.Get(ctx, r.Client, cluster.Spec.ControlPlaneRef, cluster.Namespace)
-		switch {
-		case apierrors.IsNotFound(errors.Cause(err)):
+		if s.controlPlaneIsNotFound {
 			// All good - the control plane resource has been deleted
 			conditions.MarkFalse(cluster, clusterv1.ControlPlaneReadyCondition, clusterv1.DeletedReason, clusterv1.ConditionSeverityInfo, "")
-		case err != nil:
-			return reconcile.Result{}, errors.Wrapf(err, "failed to get %s %q for Cluster %s/%s",
-				path.Join(cluster.Spec.ControlPlaneRef.APIVersion, cluster.Spec.ControlPlaneRef.Kind),
-				cluster.Spec.ControlPlaneRef.Name, cluster.Namespace, cluster.Name)
-		default:
-			// Report a summary of current status of the control plane object defined for this cluster.
-			conditions.SetMirror(cluster, clusterv1.ControlPlaneReadyCondition,
-				conditions.UnstructuredGetter(obj),
-				conditions.WithFallbackValue(false, clusterv1.DeletingReason, clusterv1.ConditionSeverityInfo, ""),
-			)
+		}
 
+		if s.controlPlane != nil {
 			// Issue a deletion request for the control plane object.
 			// Once it's been deleted, the cluster will get processed again.
-			if err := r.Client.Delete(ctx, obj); err != nil {
+			if err := r.Client.Delete(ctx, s.controlPlane); err != nil {
```

Comment:
We should add an if around it to only execute Delete if DeletionTImestamp.IsZero. We should also ignore NotFound errors from the Delete call